src/main/java/io/cdap/plugin/gcp/dataplex/source/config/DataplexBatchSourceConfig.java

```diff
@@ -0,0 +1,301 @@
+            try {
+              Entity entityBean = dataplexInterface.getEntity(getCredentials(), tryGetProject(), location,
+                lake, zone, entity);
+              return entityBean;
+            } catch (DataplexException e) {
+              configureDataplexException(entity, NAME_ENTITY, e, collector);
+              return null;
+            }
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  public void validateBigQueryDataset(FailureCollector collector, String project, String dataset, String tableName) {
+    Table sourceTable = BigQueryUtil.getBigQueryTable(project, dataset, tableName, this.getServiceAccount(),
+      this.isServiceAccountFilePath(), collector);
+    if (sourceTable == null) {
+      return;
+    }
+    if (sourceTable.getDefinition() instanceof StandardTableDefinition) {
+      TimePartitioning timePartitioning = ((StandardTableDefinition) sourceTable.getDefinition()).getTimePartitioning();
+      if (timePartitioning == null) {
+        return;
+      }
+    }
+
+    String partitionFromDate = getPartitionFrom();
+    String partitionToDate = getPartitionTo();
+
+    if (partitionFromDate == null && partitionToDate == null) {
+      return;
+    }
+    LocalDate fromDate = null;
+    if (partitionFromDate != null) {
+      try {
+        fromDate = LocalDate.parse(partitionFromDate);
+      } catch (DateTimeException ex) {
+        collector.addFailure("Invalid partition from date format.",
+          "Ensure partition from date is of format 'yyyy-MM-dd'.")
+          .withConfigProperty(BigQuerySourceConfig.NAME_PARTITION_FROM);
+      }
+    }
+    LocalDate toDate = null;
+    if (partitionToDate != null) {
+      try {
+        toDate = LocalDate.parse(partitionToDate);
+      } catch (DateTimeException ex) {
+        collector.addFailure("Invalid partition to date format.", "Ensure partition to date is of format 'yyyy-MM-dd'.")
+          .withConfigProperty(BigQuerySourceConfig.NAME_PARTITION_TO);
+      }
+    }
+
+    if (fromDate != null && toDate != null && fromDate.isAfter(toDate) && !fromDate.isEqual(toDate)) {
+      collector.addFailure("'Partition From Date' must be before or equal 'Partition To Date'.", null)
+        .withConfigProperty(BigQuerySourceConfig.NAME_PARTITION_FROM)
+        .withConfigProperty(BigQuerySourceConfig.NAME_PARTITION_TO);
+    }
+  }
+
+  private void configureDataplexException(String dataplexConfigProperty, String dataplexConfigPropType,
+                                          DataplexException e,
+                                          FailureCollector failureCollector) {
+    if (("404").equals(e.getCode())) {
+      failureCollector
+        .addFailure("'" + dataplexConfigProperty + "' could not be found. Please ensure that it exists in " +
+          "Dataplex.", null).withConfigProperty(dataplexConfigPropType);
+    } else {
+      failureCollector.addFailure(e.getCode() + ": " + e.getMessage(), null)
```

Comment:
Can we have a better error msg, similar to the 404 case above?

Reply:
done. WIll move the sink config code to base config and will reuse it here also